### PR TITLE
Add Number Pad Hpah 4x5

### PR DESCRIPTION
**Pull request by vuhopkep**
Added files for Hpah PCB

### CODE REVIEW

**Comment by drashna** [comment]
Oh, and please make sure that all of the files are all lower case. 

Because we have a lot of Windows and Linux users, it helps to prevent any case sensitivity issues.

File: keyboards/Hpah/rules.mk [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+#   Teensy halfKay   512
+#   Teensy++ halfKay 1024
+#   Atmel DFU loader 4096
+#   LUFA bootloader  4096
+#   USBaspLoader     2048
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
[the remark this answers is not in the document]
Comment (vuhopkep) [reply keyboards/Hpah/rules.mk]: I'm not using develop board like pro micro or teensy.
I using QMK DFU bootloader on this board. 
sorry for late response and my bad english. 
This is first time i using github, i'm so confused. Thank you for the best supporting 
Here is some picture about hpah pcb: 
https://www.flickr.com/photos/128706535@N02/albums/72157699051984171
https://www.youtube.com/watch?v=weOwEu7B81E&t=

[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/Hpah/rules.mk]: Not a problem! in that case this setting is fine.  But you may want to use `BOOTLOADER = qmk-dfu` here.  

For the dfu bootloaders, it doesn't matter as much for this, but the pro micros and teensy's have special handling.  

So you're good here.

File: keyboards/Hpah/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+   KEYMAP(
+		KC_NLCK, KC_PSLS, KC_PAST, LT(1, KC_PMNS), 
+		KC_P7, KC_P8, KC_P9, KC_PMNS, 
+		KC_P4, KC_P5, KC_P6, KC_PPLS, 
+		KC_P1, KC_P2, KC_P3, KC_ENT, 
+		KC_P0, KC_PDOT, KC_PDOT, KC_ENT),
+
+	KEYMAP(
```
Comment (drashna) [inline keyboards/Hpah/keymaps/default/keymap.c]:
```suggestion
	LAYOUT_ortho_5x4(
```

File: keyboards/Hpah/readme.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+Keyboard Maintainer: [Hnah](https://github.com/vuhopkep) 
+Hardware Supported: Hpah PCB   
+
+
+Make example for this keyboard (after setting up your build environment):
+
+    make Hpah:default
+
+See [build environment setup](https://docs.qmk.fm/build_environment_setup.html) then the [make instructions](https://docs.qmk.fm/make_instructions.html) for more information.
```
Comment (drashna) [inline keyboards/Hpah/readme.md]: Could you replace this link block with this, instead: 

```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/Hpah/rules.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/Hpah/rules.mk]:
```suggestion
BOOTLOADER = qmk-dfu
```

File: keyboards/Hpah/rules.mk [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/Hpah/rules.mk]:
```suggestion
LAYOUTS = ortho_5x4

# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
```

**Comment by drashna** [comment]
Just wanted to touch base here.  

Did you get a chance to update this?

**Comment by mechmerlin** [comment]
Hi @vuhopkep have you had time to work on this?

**Comment by drashna** [comment]
@vuhopkep Have you had a chance to update this?

**Comment by vuhopkep** [comment]
I'm really sorry for very late respon.
I'm working now for change.
I'm so sorry!

File: keyboards/Hpah/Hpah.c [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#include "Hpah.h"
+
+void matrix_init_kb(void) {
+	matrix_init_user();
```
Comment (drashna) [inline keyboards/Hpah/Hpah.c]:
```suggestion
    setPinOutput(F0);
	  matrix_init_user();
```

File: keyboards/Hpah/Hpah.c [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+
+void led_set_user(uint8_t usb_led) {
+
+	if (usb_led & (1 << USB_LED_NUM_LOCK)) {
```
Comment (drashna) [inline keyboards/Hpah/Hpah.c]:
```suggestion
    if (IS_LED_ON(usb_led, USB_LED_NUM_LOCK)) {
```

File: keyboards/Hpah/Hpah.c [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+		DDRF |= (1 << 0); PORTF &= ~(1 << 0);
```
Comment (drashna) [inline keyboards/Hpah/Hpah.c]:
```suggestion
		writePinLow(F0);
```

File: keyboards/Hpah/Hpah.c [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+	} else {
+		DDRF &= ~(1 << 0); PORTF &= ~(1 << 0);
```
Comment (drashna) [inline keyboards/Hpah/Hpah.c]:
```suggestion
		writePinHigh(F0);
```

File: keyboards/Hpah/Hpah.h [hunk 9]
```diff
@@ -0,0 +1,20 @@
+#ifndef HPAH_H
```
Comment (drashna) [inline keyboards/Hpah/Hpah.h]:
```suggestion
#pragma once
```

File: keyboards/Hpah/Hpah.h [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+#define HPAH_H
```
Comment (drashna) [inline keyboards/Hpah/Hpah.h]:
```suggestion
```

File: keyboards/Hpah/Hpah.h [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+	K40, K41, K42, K43  \
+) { \
+	{ K00,   K01,   K02,   K03 }, \
+	{ K10,   K11,   K12,   K13 }, \
+	{ K20,   K21,   K22,   K23 }, \
+	{ K30,   K31,   K32,   K33 }, \
+	{ K40,   K41,   K42,   K43 }  \
+}
+
+#endif
```
Comment (drashna) [inline keyboards/Hpah/Hpah.h]:
```suggestion
```

File: keyboards/Hpah/keymaps/default/rules.mk [hunk 12]
```diff
@@ -0,0 +1,3 @@
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/Hpah/keymaps/default/rules.mk]:
```suggestion
```

File: keyboards/Hpah/keymaps/default/rules.mk [hunk 13]
```diff
@@ -0,0 +1,3 @@
...
+	include ../../../../Makefile
```
Comment (drashna) [inline keyboards/Hpah/keymaps/default/rules.mk]:
```suggestion
```

File: keyboards/Hpah/keymaps/default/rules.mk [hunk 14]
```diff
@@ -0,0 +1,3 @@
...
+endif
```
Comment (drashna) [inline keyboards/Hpah/keymaps/default/rules.mk]:
```suggestion
```

File: keyboards/Hpah/Hpah.c [hunk 15]
```diff
@@ -0,0 +1,16 @@
+#include "Hpah.h"
```
Comment (drashna) [inline keyboards/Hpah/Hpah.c]:
```suggestion
#include "hpah.h"
```

**Comment by drashna** [comment]
Could you  change the folder and file names to be lower case, too?

File: keyboards/Hpah/Hpah.c [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+
+void led_set_user(uint8_t usb_led) {
+	if (IS_LED_ON(usb_led, USB_LED_NUM_LOCK)) {
+		writePinLow(F0);
+	} else {
+		writePinHigh(F0);
+	}
+}
```
Comment (drashna) [inline keyboards/Hpah/Hpah.c]: Well, in the meanwhile, how this works has changed, actually: 
https://docs.qmk.fm/#/custom_quantum_functions?id=led-control

File: keyboards/Hpah/config.h [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+#define MATRIX_COLS 4
+
+/* pin-out */
+#define MATRIX_ROW_PINS { B3, B0, F7, F6, F5 }
+#define MATRIX_COL_PINS { B2, B1, B7, D0 }
+#define UNUSED_PINS
+
+/* ws2812 RGB LED */
+#define RGB_DI_PIN E6
+#define RGBLIGHT_TIMER
```
Comment (drashna) [inline keyboards/Hpah/config.h]:
```suggestion
```

File: keyboards/Hpah/rules.mk [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+#     clock generation for the USB and other AVR subsections. Do NOT tack on a 'UL'
+#     at the end, this will be done automatically to create a 32-bit value in your
+#     source code.
+#
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
```
Comment (drashna) [inline keyboards/Hpah/rules.mk]: This entire section is no longer needed

**Pull request closed without merging**